plugins/net_plugin/net_plugin.cpp

```diff
@@ -1978,51 +1980,49 @@ namespace eosio {
 
    void dispatch_manager::retry_fetch(const connection_ptr& c) {
       fc_dlog( logger, "retry fetch" );
-      std::unique_lock<std::mutex> g_c_conn( c->conn_mtx );
-      if( !c->last_req ) {
-         return;
-      }
-      fc_wlog( logger, "failed to fetch from ${p}", ("p", c->peer_address()) );
+      request_message last_req;
       block_id_type bid;
-      if( c->last_req->req_blocks.mode == normal && !c->last_req->req_blocks.ids.empty() ) {
-         bid = c->last_req->req_blocks.ids.back();
-      } else {
-         fc_wlog( logger, "no retry, block mpde = ${b} trx mode = ${t}",
-                  ("b", modes_str( c->last_req->req_blocks.mode ))( "t", modes_str( c->last_req->req_trx.mode ) ) );
-         return;
+      {
+         std::lock_guard<std::mutex> g_c_conn( c->conn_mtx );
+         if( !c->last_req ) {
+            return;
+         }
+         fc_wlog( logger, "failed to fetch from ${p}", ("p", c->peer_address()) );
+         if( c->last_req->req_blocks.mode == normal && !c->last_req->req_blocks.ids.empty() ) {
+            bid = c->last_req->req_blocks.ids.back();
+         } else {
+            fc_wlog( logger, "no retry, block mpde = ${b} trx mode = ${t}",
+                     ("b", modes_str( c->last_req->req_blocks.mode ))( "t", modes_str( c->last_req->req_trx.mode ) ) );
+            return;
+         }
+         last_req = *c->last_req;
       }
-      g_c_conn.unlock();
-      std::shared_lock<std::shared_mutex> g( my_impl->connections_mtx );
-      for( auto& conn : my_impl->connections ) {
-         if( conn == c ) continue;
-
-         std::unique_lock<std::mutex> g_conn_conn( conn->conn_mtx );
-         if( conn->last_req ) {
-            continue;
+      my_impl->for_each_connection( [this, &c, &last_req, &bid]( auto& conn ) {
+         if( conn == c )
+            return true;
+         {
+            std::lock_guard<std::mutex> guard( conn->conn_mtx );
+            if( conn->last_req ) {
+               return true;
+            }
```

Comment:
What was this code originally doing?  I don't see  (either in original or new) any effect it has.  Will the above lambda return true or false at the end of the Lambda?

Reply:
The end of the lambda is below and it returns true.

This requests from a diff connection since this connection was closed.